build/source/driver/multi_driver.f90

```diff
@@ -567,216 +646,231 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
+    case(ixRestart_id);    printRestart = (timeStruct%var(iLookTIME%ih) == 0 .and. timeStruct%var(iLookTIME%imin) == 0)
+    case(ixRestart_never); printRestart = .false.
+    case default; call handle_err(20,'unable to identify option for the restart file')
+   end select
+ 
+   ! set the flag to compute the vegetation flux
+   computeVegFluxFlag = (computeVegFlux(iGRU)%hru(iHRU) == yes)
+ 
+   ! run the model for a single parameter set and time step
+   call coupled_em(&
+                   ! model control
+                   istep,                          & ! intent(in):    time step index
+                   printRestart,                   & ! intent(in):    flag to print a re-start file
+                   output_fileSuffix,              & ! intent(in):    name of the experiment used in the restart file
+                   dt_init(iGRU)%hru(iHRU),        & ! intent(inout): initial time step
+                   computeVegFluxFlag,             & ! intent(inout): flag to indicate if we are computing fluxes over vegetation (.false. means veg is buried with snow)
+                   ! data structures (input)
+                   timeStruct,                     & ! intent(in):    model time data
+                   typeStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    local classification of soil veg etc. for each HRU
+                   attrStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    local attributes for each HRU
+                   forcStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    model forcing data
+                   mparStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    model parameters
+                   bvarStruct%gru(iGRU),           & ! intent(in):    basin-average model variables
+                   ! data structures (input-output)
+                   indxStruct%gru(iGRU)%hru(iHRU), & ! intent(inout): model indices
+                   progStruct%gru(iGRU)%hru(iHRU), & ! intent(inout): model prognostic variables for a local HRU
+                   diagStruct%gru(iGRU)%hru(iHRU), & ! intent(inout): model diagnostic variables for a local HRU
+                   fluxStruct%gru(iGRU)%hru(iHRU), & ! intent(inout): model fluxes for a local HRU
+                   ! error control
+                   err,message)            ! intent(out): error control
+   call handle_err(err,message)
+  
+   ! save the flag for computing the vegetation fluxes
+   if(computeVegFluxFlag)      computeVegFlux(iGRU)%hru(iHRU) = yes
+   if(.not.computeVegFluxFlag) computeVegFlux(iGRU)%hru(iHRU) = no
+
+   kHRU = 0
+   ! identify the downslope HRU
+   do jHRU=1,nHRU
+    if(typeStruct%gru(iGRU)%hru(iHRU)%var(iLookTYPE%downHRUindex) == typeStruct%gru(iGRU)%hru(jHRU)%var(iLookTYPE%hruIndex))then
+     if(kHRU==0)then  ! check there is a unique match
+      kHRU=jHRU
+     else
+      call handle_err(20,'multi_driver: only expect there to be one downslope HRU')
```

Comment:
Move check to much earlier in the code. There is no reason to run the model and wait till after the first to call to `coupled_em`to find that there is a problem with the topology. Move check to initialization of GRU and HRU structures.